### DISCUSSION TITLE
GS1 Data Matrix - Unable to generate without AI 01

### DISCUSSION THREAD

**Discussion by umich-devin** [metafloor/bwip-js#339] [Q&A] [answered]
Good Afternoon,

A previous version of bwip-js (from last summer) allowed us to generate a GS1 Data Matrix without the (01) AI. This appears to have changed since then. Was this intentional, and is there any easy workaround?

Thank you,

Devin

**Comment by metafloor** [comment, marked as answer]
Yes, the upstream BWIPP code has added full GS1 spec-conforming functionality.  If it's a GS1 barcode, then it needs an AI.  If you don't want the AI, use the base datamatrix encoder.

**Comment by umich-devin** [comment]
Thank you for the quick clarification, we are using other AIs, just not 01 in rare circumstances where a GTIN doesn't exist. We will find something else for this. Again - thank you!

**Comment by terryburton** [comment]
The `gs1datamatrix` encoder will certainly allow you to generate a GS1 Data Matrix without an AI (01).

If it is rejecting your input then it will report exactly which other AI has a mandatory dependency on AI (01), e.g. `One of more requisite AIs for AI (17) are missing`.

If the mandatory associations between AIs are not satisfied, then the AI data is invalid and such symbols should not be used between trade partners.

**Reply by umich-devin** [reply]
Hi Terry,

Thank you for the follow-up. I appreciate the understandable explanation. We wish that all was true for all vendor products, but that is unfortunately not always the case with what we receive. We will find a workaround.

Thank you,

Devin

**Reply by terryburton** [reply]
The `dontlint` option disables GS1 data validation which would allow you to continue to use the application-specific encoder.

**Reply by umich-devin** [reply]
> The `dontlint` option disables GS1 data validation which would allow you to continue to use the application-specific encoder.

Thank you! This is very helpful for our /unfortunate/ use case.
